jib-core/src/main/java/com/google/cloud/tools/jib/image/json/ContainerConfigurationTemplate.java

```diff
@@ -161,14 +166,18 @@ public void addLayerDiffId(DescriptorDigest diffId) {
   }
 
   @Nullable
-  ImmutableList<String> getContainerExposedPorts() {
+  ImmutableList<PortsWithProtocol> getContainerExposedPorts() {
     // TODO: Do this conversion somewhere else
     if (config.ExposedPorts == null) {
       return null;
     }
-    ImmutableList.Builder<String> ports = new ImmutableList.Builder<>();
+    ImmutableList.Builder<PortsWithProtocol> ports = new ImmutableList.Builder<>();
     for (Map.Entry<String, Map<?, ?>> entry : config.ExposedPorts.entrySet()) {
-      ports.add(entry.getKey());
+      List<String> split = Splitter.on('/').splitToList(entry.getKey());
```

Comment:
This can use the same parsing mechanism as in `ExposedPortsParser`

Reply:
If you mean using pattern matching, I found this to be much simpler, since we'll never encounter ranges here.

Reply:
What if the config ports is not in the assumed format? Like `split.get(0)` could be not a parseable integer.

Reply:
I guess at that point the question becomes how do we handle a malformed container config in general, though? The Go Container type explicitly states that the exposed ports will be in this format.

Reply:
It looks like we need to figure out what the spec allows, and handle that (whether it is ranges or not). Once we know that we can fail hard if the image format is invalid.

Reply:
I gave it a try and created a container with a range of exposed ports in the container config:

```
$ docker inspect gcr.io/tcordle-test/hellogradle
...
    "Config": {
        ...
        "ExposedPorts": {
            "1000-1003/udp": {}
        },
        ...
    }
...
```

and then I tried running the container with the -P flag to automatically create port mappings, and I got this error:

```
$ docker run -P gcr.io/tcordle-test/hellogradle
docker: Error response from daemon: driver failed programming external connectivity 
on endpoint affectionate_agnesi8 
(34e06094483a6871f8257c2a26d77935defbb91362fe041c33035602b4f2645b):  
(iptables failed: iptables --wait -t nat -A DOCKER -p udp -d 0/0 --dport 32787 -j DNAT --to-destination 192.168.9.2:0 ! -i docker0: iptables v1.6.1: Port `0' not valid
```

Doesn't look like ranges are supported within the container config.

Reply:
Yea, I think even if they were we should still follow the spec stated in https://github.com/opencontainers/image-spec/blob/master/config.md#properties, which only allows single ports.